src/common.js

```diff
@@ -12,13 +12,18 @@ if (typeof location !== 'undefined') {
 }
 
 const backslashRegEx = /\\/g;
+const protocolre = /^[a-z][a-z0-9.+-]*\:/i
 export function resolveIfNotPlainOrUrl (relUrl, parentUrl) {
   if (relUrl.indexOf('\\') !== -1)
     relUrl = relUrl.replace(backslashRegEx, '/');
   // protocol-relative
   if (relUrl[0] === '/' && relUrl[1] === '/') {
     return parentUrl.slice(0, parentUrl.indexOf(':') + 1) + relUrl;
   }
+  // non-relative URL with protocol
+  else if (protocolre.test(relUrl)) {
+    return relUrl;
+  }
```

Comment:
Thanks so much for posting this. The fix is right, but I would rather move this check to just before its usage in the import maps implementation rather. This resolve function was carefully designed around the WhatWG specification of relative URL matching - I'm actually against supporting maps for URLs so am still secretly hoping this gets cut from the spec for a better approach of handling the polyfill codepaths it was created for (ie this project!).

Reply:
Hi @guybedford Do you think this check belongs in the `resolve` function:

```js
const protocolre = /^[a-z][a-z0-9.+-]*\:/i
async function resolve (id, parentUrl) {
  if (protocolre.test(id)) {
    return Promise.resolve(id);
  }
  parentUrl = parentUrl || pageBaseUrl;

  if (importMapPromise)
    return importMapPromise
    .then(function () {
      return resolveImportMap(id, parentUrl, importMap);
    });

  return resolveImportMap(id, parentUrl, importMap);
}
```

Does it look like this is the equivalent for system.js: https://github.com/systemjs/systemjs/blob/60cd9e2c3d2d7c3765c9afa10b5bd205ed7a71d9/src/features/basic-resolve.js#L5 ?

Reply:
Although it looks like the resolve is the same as when using importMaps: https://github.com/systemjs/systemjs/blob/60cd9e2c3d2d7c3765c9afa10b5bd205ed7a71d9/src/features/import-map.js#L39

Reply:
I would include it as a check in the resolveImportMap function if the return value from `resolveIfNotPlainOrURL` is not defined - https://github.com/systemjs/systemjs/blob/60cd9e2c3d2d7c3765c9afa10b5bd205ed7a71d9/src/common.js#L162.

Reply:
so like https://glitch.com/edit/#!/bramble-veil?path=shimport/es-module-shims-temp.js:165:16 ?

Reply:
Yes exactly!

Reply:
Fixed!